### PR TITLE
Refactor dead sections

### PR DESCRIPTION
**Pull request by Castavo**
linked to: #4660 

In this PR there is, for now
* Removal of an unused field (planned to use it but it got scrapped)
* Renaming of classes and attributes

### CODE REVIEW

**Review by clarani — COMMENTED** [review]
Great PR, thank you :pray:
I can approve the frontend part once the PR is ready (I will have to test)

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackProperties.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ interface TrackProperties {
     fun getTrackChunkGradient(trackChunk: DirTrackChunkId): DistanceRangeMap<Double>
     fun getTrackChunkLoadingGaugeConstraints(trackChunk: TrackChunkId): DistanceRangeMap<LoadingGaugeConstraint>
     fun getTrackChunkCatenaryVoltage(trackChunk: TrackChunkId): DistanceRangeMap<String>
-    fun getTrackChunkDeadSection(trackChunk: DirTrackChunkId): DistanceRangeMap<DeadSection>
+    fun getTrackChunkNeutralSection(trackChunk: DirTrackChunkId): DistanceRangeMap<NeutralSection>
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackProperties.kt]:
```suggestion
    fun getTrackChunkNeutralSections(trackChunk: DirTrackChunkId): DistanceRangeMap<NeutralSection>
```

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt [hunk 2]
```diff
@@ -201,8 +201,8 @@ class RawInfraImpl(
         return trackChunkPool[trackChunk].catenaryVoltage
     }
 
-    override fun getTrackChunkDeadSection(trackChunk: DirTrackChunkId): DistanceRangeMap<DeadSection> {
-        return trackChunkPool[trackChunk.value].deadSections.get(trackChunk.direction)
+    override fun getTrackChunkNeutralSection(trackChunk: DirTrackChunkId): DistanceRangeMap<NeutralSection> {
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt]:
```suggestion
    override fun getTrackChunkNeutralSections(trackChunk: DirTrackChunkId): DistanceRangeMap<NeutralSection> {
```

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSInfra.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class RJSInfra {
             .build()
             .adapter(RJSInfra.class);
 
-    public static final transient String CURRENT_VERSION = "3.3.0";
+    public static final transient String CURRENT_VERSION = "3.6.0";
```
Comment (multun) [inline core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSInfra.java]: That's quite a bump, why is that? I'd rather have a single version bump and migration

File: editoast/src/tests/small_infra/small_infra.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-    "version": "3.3.1",
+    "version": "3.6.0",
```
Comment (multun) [inline editoast/src/tests/small_infra/small_infra.json]: This will need a rebase once https://github.com/osrd-project/osrd/pull/4728 is merged

**Review by multun — CHANGES_REQUESTED** [review]

**Comment by multun** [comment]
A very nice PR, thanks :) Please change it as you see fit, and ask @flomonster to merge it you think it's ready (all core maintainers are away next week)

**Review by flomonster — APPROVED** [review]
Thanks for this PR!
LGTM

**Review by clarani — APPROVED** [review]
Front looks good to me, tested in local :heavy_check_mark:

**Review by younesschrifi — APPROVED** [review]
LGTM

**Pull request merged**